### PR TITLE
fix(gridnav): fix the `lv_gridnav_remove` which may remove the event incorrectly

### PR DESCRIPTION
**Pull request by TCRnext**
### Description of the feature or fix
图片
In function void lv_gridnav_remove(lv_obj_t * obj) if the obj has no event which has its callback function is gridnav_event_cb, but the obj has at least one event.The function will delete the last event of the obj and free its userdata.In some case,it's will course immediately crash.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html)
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Comment by C47D** [comment]
Fixes #5409

File: src/others/gridnav/lv_gridnav.c [hunk 1]
```diff
@@ -85,9 +85,11 @@ void lv_gridnav_remove(lv_obj_t * obj)
     }
 
     if(event_dsc) {
-        lv_free(lv_event_dsc_get_user_data(event_dsc));
-        lv_obj_remove_event(obj, i);
+        if(lv_event_dsc_get_cb(event_dsc) == gridnav_event_cb) {
```
Comment (PGNetHun) [inline src/others/gridnav/lv_gridnav.c]: I think it’s better to move the code inside the loop `if` statement, above the `break;` line, it’s simpler, result is same, but saves us one more comparison.

**Review by PGNetHun — CHANGES_REQUESTED** [review]

File: src/others/gridnav/lv_gridnav.c (hunk 1 above)
Comment (kisvegabor) [reply src/others/gridnav/lv_gridnav.c]: I agree.

**Review by kisvegabor — CHANGES_REQUESTED** [review]

**Review by kisvegabor — DISMISSED** [review]
Thank you!

**Review by kisvegabor — APPROVED** [review]

**Pull request merged**